tests/utils/subnet.go

```diff
@@ -105,15 +167,42 @@ func GetDefaultChainURI(blockchainID string) string {
 	return fmt.Sprintf("%s/ext/bc/%s/rpc", DefaultLocalNodeURI, blockchainID)
 }
 
-// RunDefaultHardhatTests runs the hardhat tests on a given blockchain ID
-// with default parameters. Default parameters are:
-// 1. Hardhat contract environment is located at ./contracts
-// 2. Hardhat test file is located at ./contracts/test/<test>.ts
-// 3. npx is available in the ./contracts directory
-func RunDefaultHardhatTests(ctx context.Context, blockchainID string, test string) {
+// CreateAndRunHardhatTests creates a subnet and blockchain and then
+// runs the hardhat tests on the new blockchain with default parameters.
+//
+//	Default parameters are:
+//
+// 1. Genesis file is located at ./tests/precompile/genesis/<test>.json
+// 2. Hardhat contract environment is located at ./contracts
+// 3. Hardhat test file is located at ./contracts/test/<test>.ts
+// 4. npx is available in the ./contracts directory
+func CreateAndRunHardhatTests(ctx context.Context, test string) {
```

Comment:
Seems like we're moving towards a point where we should stop making these assumptions and pass them in from the caller.